source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2515 @@
+  cluster backed by replica sets). If this field is omitted, it should be
+  assumed that there is no topology requirement for the test.
+
+  When matching a "sharded-replicaset" topology, test runners MUST ensure that
+  all shards are backed by a replica set. The process for doing so is described
+  in `Determining if a Sharded Cluster Uses Replica Sets`_.
+
+
+entity
+~~~~~~
+
+An entity (e.g. client, collection, session object) that will be created in the
+`Entity Map`_ before each test is executed.
+
+This document MUST contain **exactly one** top-level key that identifies the
+entity type and maps to a nested document, which specifies a unique name for the
+entity (``id`` key) and any other parameters necessary for its construction.
+Tests SHOULD use sequential names based on the entity type (e.g. "session0",
+"session1").
+
+When defining an entity document in YAML, a `node anchor`_ SHOULD be created on
+the entity's ``id`` key. This anchor will allow the unique name to be referenced
+with an `alias node`_ later in the file (e.g. from another entity or
+`operation`_ document) and also leverage YAML's parser for reference validation.
+
+.. _node anchor: https://yaml.org/spec/1.2/spec.html#id2785586
+.. _alias node: https://yaml.org/spec/1.2/spec.html#id2786196
+
+The structure of this document is as follows:
+
+.. _entity_client:
+
+- ``client``: Optional document. Corresponds with a MongoClient object.
+
+  The structure of this document is as follows:
+
+  - ``id``: Required string. Unique name for this entity. The YAML file SHOULD
+    define a `node anchor`_ for this field (e.g. ``id: &client0 client0``).
+
+  - ``uriOptions``: Optional document. Additional URI options to apply to the
+    test suite's connection string that is used to create this client. Any keys
+    in this document MUST override conflicting keys in the connection string.
+
+    Documentation for supported options may be found in the
+    `URI Options <../uri-options/uri-options.rst>`__ spec, with one notable
+    exception: if ``readPreferenceTags`` is specified in this document, the key
+    will map to an array of strings, each representing a tag set, since it is
+    not feasible to define multiple ``readPreferenceTags`` keys in the document.
+
+  .. _entity_client_useMultipleMongoses:
+
+  - ``useMultipleMongoses``: Optional boolean. If true and the topology is a
+    sharded cluster, the test runner MUST assert that this MongoClient connects
+    to multiple mongos hosts (e.g. by inspecting the connection string). If
+    false and the topology is a sharded cluster, the test runner MUST ensure
+    that this MongoClient connects to only a single mongos host (e.g. by
+    modifying the connection string). This option has no effect for non-sharded
+    topologies.
+
+  .. _entity_client_observeEvents:
+
+  - ``observeEvents``: Optional string or array of strings. One or more types of
```

Comment:
This section reads a little confusing, the first line states `Optional string or array of strings.` and then toward the end of this bullet `Supported types correspond to those documented in `expectedEvent`_ and are as follows:`, this leads me to believe that `observeEvents` can either be a string, array of strings, or `expectedEvents` object? Why is the `expectedEvents` included in this section?

Reply:
I have some troubles differentiating event in terms of a changeStream event and command monitoring / outgoing command event. Maybe these shouldn't be under the same umbrella (property)? Or perhaps I'm not fully understanding the cases here.

Reply:
I think I'd just like a line here or two to specify the kind of string. For better Illustration:

> Optional string or array of strings.

Toward to top of the description I think it may be a string or array of strings matching the string values for `commandStartedEvent`, `commandSucceededEvent`, `commandFailedEvent`. 

```json
{
  "observeEvents": "commandStartedEvent"
}
```

or 

```json
{
  "observeEvents": ["commandStartedEvent"]
}
```

But then toward the bottom:

> Supported types correspond to those documented in `expectedEvent`_ and are
    as follows:

When I see the "supported types" link to the `commandStartedEvent` object. I think that the "supported type" can be one of these objects.

```json
{
  "observeEvents": {
    "commandStartedEvent": {}
  }
}
```

Or some combination of supporting both.

After reading this now I think I believe it as the first example, but just some differentiation between event object type and event key would add clarity here.

Reply:
Rewriting as:

> Supported types correspond to the top-level keys documented in `expectedEvent`_ and are as follows:

Will resolve once pushed.